src/catalog/types.ts

```diff
@@ -295,6 +295,16 @@ export interface CommitTableResponse {
   metadata: TableMetadata
 }
 
+/**
+ * Response from the catalog configuration endpoint (GET /v1/config).
+ */
+export interface CatalogConfig {
+  defaults: Record<string, string>
+  overrides: Record<string, string>
+  endpoints?: string[]
+  'idempotency-key-lifetime'?: string
```

Comment:
The implementation for actually assigning idempotency-keys to various namespace operation requests and table operation requests has not been completed. I would like to make it out of scope.